### PR TITLE
Milight: Move simulated V6 bridge to own bundle. Use OSGi DS annotations

### PR DESCRIPTION
**Pull request by davidgraeff**
Signed-off-by: David Gräff <david.graeff@web.de>

### CODE REVIEW

File: addons/binding/pom.xml [hunk 1]
```diff
@@ -53,6 +53,7 @@
     <module>org.openhab.binding.miele</module>
     <module>org.openhab.binding.mihome</module>
     <module>org.openhab.binding.milight</module>
+    <module>org.openhab.binding.milightdev</module>
```
Comment (martinvw) [inline addons/binding/pom.xml]: @kaikreuzer is this the / a proper package?

File: addons/binding/org.openhab.binding.milightdev/src/main/java/org/openhab/binding/milight/internal/simulatedbridge/TestDiscovery.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            server = new EmulatedV6Bridge();
+        } catch (Exception e) {
+            logger.warn("An error occurred", e);
+        }
+    }
+
+    @Override
+    protected void startScan() {
+    }
+}
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.milightdev/src/main/java/org/openhab/binding/milight/internal/simulatedbridge/TestDiscovery.java]: It seems you did not commit the actual removal, now they are just copies, see also:

https://github.com/davidgraeff/openhab2-addons/tree/milightfix/addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/test

Comment (martinvw) [inline addons/binding/org.openhab.binding.milightdev/src/main/java/org/openhab/binding/milight/internal/simulatedbridge/TestDiscovery.java]: Can you add a newline :-)

File: addons/binding/org.openhab.binding.milightdev/pom.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.openhab.binding</groupId>
+    <artifactId>pom</artifactId>
+    <version>2.2.0-SNAPSHOT</version>
+  </parent>
+
+  <properties>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.milightdev/pom.xml]: These are not actually needed, or at least should not be needed

File: addons/binding/org.openhab.binding.milightdev/about.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+<p>If you did not receive this Content directly from the openHAB community, the Content is 
+being redistributed by another party (&quot;Redistributor&quot;) and different terms and conditions may
+apply to your use of any object code in the Content.  Check the Redistributor's license that was 
+provided with the Content.  If no such license exists, contact the Redistributor.  Unless otherwise
+indicated below, the terms and conditions of the EPL still apply to any source code in the Content
+and such source code may be obtained at <a href="http://www.openhab.org/">openhab.org</a>.</p>
+
+</body>
+</html>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.milightdev/about.html]: newline :-)

File: addons/binding/org.openhab.binding.milightdev/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+target/
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.milightdev/.gitignore]: Can this be removed

**Comment by martinvw** [comment]
```
10:33:00.826  |  [INFO] ------------------------------------------------------------------------
10:33:00.835  |  [ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.0.0:compile (default-compile) on project org.openhab.binding.milightdev: Compilation failure: Compilation failure:
10:33:00.835  |  [ERROR] /scratch/jenkins/workspace/PR-openHAB2-Addons/addons/binding/org.openhab.binding.milightdev/src/main/java/org/openhab/binding/milight/internal/simulatedbridge/EmulatedV6Bridge.java:[120]
10:33:00.835  |  [ERROR] DatagramSocket datagramSocket = new DatagramSocket(MilightBindingConstants.PORT_DISCOVER);
10:33:00.836  |  [ERROR] ^^^^^^^^^^^^^^^^^^^^^^^
10:33:00.836  |  [ERROR] MilightBindingConstants cannot be resolved to a variable
10:33:00.836  |  [ERROR] /scratch/jenkins/workspace/PR-openHAB2-Addons/addons/binding/org.openhab.binding.milightdev/src/main/java/org/openhab/binding/milight/internal/simulatedbridge/EmulatedV6Bridge.java:[157]
10:33:00.836  |  [ERROR] DatagramSocket datagramSocket = new DatagramSocket(MilightBindingConstants.PORT_VER6);
10:33:00.836  |  [ERROR] ^^^^^^^^^^^^^^^^^^^^^^^
10:33:00.837  |  [ERROR] MilightBindingConstants cannot be resolved to a variable
10:33:00.837  |  [ERROR] 2 problems (2 errors)
10:33:00.837  |  [ERROR] -> [Help 1]
10:33:00.837  |  [ERROR] 
10:33:00.838  |  [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
10:33:00.838  |  [ERROR] Re-run Maven using the -X switch to enable full debug logging.
10:33:00.838  |  [ERROR] 
10:33:00.838  |  [ERROR] For more information about the errors and possible solutions, please read the following articles:
10:33:00.839  |  [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
10:33:00.839  |  [ERROR] 
10:33:00.839  |  [ERROR] After correcting the problems, you can resume the build with the command
10:33:00.840  |  [ERROR]   mvn <goals> -rf :org.openhab.binding.milightdev
10:33:01.917  |  channel stopped
```

**Comment by davidgraeff** [comment]
I have no idea why the new bundle cannot find the exported constant class of milight. Did I do something wrong in the manifests?

**Comment by martinvw** [comment]
I just realise that we might have better of if I quoted the error from travis, did not check what it said, but Jenkins does a partial build maybe it forgot the build the main project and that could be the sole reason it is missing sorry for the noice, please validate the actual problem in the travis build log.

**Comment by martinvw** [comment]
@davidgraeff ^^ it was not Jenkins only I updated my comment :-)

**Comment by davidgraeff** [comment]
And I don't know why. I export the package correctly and import it again.

**Comment by martinvw** [comment]
@davidgraeff I took the liberty to fix it, please check the commit then you will understand the problem 😜 

@kaikreuzer do you agree on the name of the artifact?

**Comment by davidgraeff** [comment]
Haha ok, eclipse removed that line apparently

**Comment by kaikreuzer** [comment]
> @kaikreuzer do you agree on the name of the artifact?

No, this will cause issues as it appears like a real binding. I would suggest to move that code into a tests bundle - it is fine to have other (non shippable) parts in those as well.

**Comment by davidgraeff** [comment]
As long as it's easily installable by non-developer users, it should be alright

**Comment by davidgraeff** [comment]
I don't know how to proceed to be honest. I want this Simulated V6 bridge to be installable by users, for the following reasons (a test bundle would is not the appropriate place therefore):

* The milight protocol changes every year at the moment: I need users with new bulbs / new milight iOS apps / new hardware bridges to install this binding and provide me with the logs.
* A user should not be required to compile this helper binding on its own, but just install it via a familiar graphical/command line interface.

**Comment by kaikreuzer** [comment]
> a test bundle would is not the appropriate place therefore

Even test bundles end up on our Artifactory (see e.g. [this one](https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/org.openhab.binding.mysensors.test/2.2.0-SNAPSHOT/)) and you can ask users simply to do a 
```
bundle:install <url>
```
in order to install it for testing. Wouldn't that suffice?

**Pull request closed without merging**